### PR TITLE
Add NativeAOT runtime pack build for iOS/tvOS/Catalyst

### PR DESCRIPTION
**Pull request by akoeplinger**
Contributes to https://github.com/dotnet/runtime/issues/81024

This is the first part that adds runtime packs to the build.

### CODE REVIEW

**Comment by jkotas** [comment]
What is going to be the name of this runtime pack?

**Comment by steveisok** [comment]
@akoeplinger official build looks good 

https://dev.azure.com/dnceng/internal/_build/results?buildId=2162257&view=results

**Review by steveisok — APPROVED** [review]
Looks good!

**Comment by jkotas** [comment]
> Similar to how we include mono in the runtime packs. Microsoft.NETCore.App.Runtime.NativeAOT.<os>-<arch>

Is the AOT compiler going to stay in Microsoft.DotNet.ILCompiler?

I am wondering whether it would make sense to do the same change for all hosts and targets.

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props [hunk 1]
```diff
@@ -174,6 +174,20 @@
     <PlatformManifestFileEntry Include="Microsoft.DiaSymReader.Native.amd64.dll" IsNative="true" FallbackFileVersion="$(MicrosoftDiaSymReaderNativeFileVersion)" />
     <PlatformManifestFileEntry Include="Microsoft.DiaSymReader.Native.arm.dll" IsNative="true" FallbackFileVersion="$(MicrosoftDiaSymReaderNativeFileVersion)" />
     <PlatformManifestFileEntry Include="Microsoft.DiaSymReader.Native.arm64.dll" IsNative="true" FallbackFileVersion="$(MicrosoftDiaSymReaderNativeFileVersion)" />
+    <!-- NativeAOT specific files -->
+    <PlatformManifestFileEntry Include="libbootstrapper.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libbootstrapperdll.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libeventpipe-disabled.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libeventpipe-enabled.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libnumasupportdynamic.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libnumasupportstatic.a" IsNative="true" />
```
Comment (am11) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:183]:
```suggestion
```
Not needed since https://github.com/dotnet/runtime/commit/de5a04b2d7d184c5a9121b56cd51c713f72def76.

```sh
$ ls /root/.nuget/packages/runtime.linux-musl-arm64.microsoft.dotnet.ilcompiler/8.0.0-preview.4.23217.4/sdk |\
     grep -vE '\.(xml|pdb)' |\
     while read name; do echo '   <PlatformManifestFileEntry Include="'$name'" IsNative="true" />'; done

   <PlatformManifestFileEntry Include="System.Private.CoreLib.dll" IsNative="true" />
   <PlatformManifestFileEntry Include="System.Private.DisabledReflection.dll" IsNative="true" />
   <PlatformManifestFileEntry Include="System.Private.Reflection.Execution.dll" IsNative="true" />
   <PlatformManifestFileEntry Include="System.Private.StackTraceMetadata.dll" IsNative="true" />
   <PlatformManifestFileEntry Include="System.Private.TypeLoader.dll" IsNative="true" />
   <PlatformManifestFileEntry Include="libRuntime.ServerGC.a" IsNative="true" />
   <PlatformManifestFileEntry Include="libRuntime.WorkstationGC.a" IsNative="true" />
   <PlatformManifestFileEntry Include="libbootstrapper.a" IsNative="true" />
   <PlatformManifestFileEntry Include="libbootstrapperdll.a" IsNative="true" />
   <PlatformManifestFileEntry Include="libeventpipe-disabled.a" IsNative="true" />
   <PlatformManifestFileEntry Include="libeventpipe-enabled.a" IsNative="true" />
   <PlatformManifestFileEntry Include="libstdc++compat.a" IsNative="true" />
```

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props [hunk 2]
```diff
@@ -174,6 +174,20 @@
...
+    <PlatformManifestFileEntry Include="libRuntime.ServerGC.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libRuntime.WorkstationGC.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libstdc++compat.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="System.Private.DisabledReflection.dll" />
+    <PlatformManifestFileEntry Include="System.Private.Reflection.Execution.dll" />
+    <PlatformManifestFileEntry Include="System.Private.StackTraceMetadata.dll" />
+    <PlatformManifestFileEntry Include="System.Private.TypeLoader.dll" />
```
Comment (marek-safar) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:190]: Why are these needed? I thought NativeAOT ships as native binary?
Is System.Private.Corelib.dll missing in the list?

**Comment by steveisok** [comment]
> I am wondering whether it would make sense to do the same change for all hosts and targets.

I think it makes sense. Should we also change the name of the mono aot compiler packs? Right now it's `Microsoft.NETCore.App.Runtime.AOT.<os>-<arch>`.

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props (hunk 2 above)
Comment (MichalStrehovsky) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:190]: These are things that could be in CoreLib but since it doesn't cost anything to have them out of CoreLib, we keep them out of CoreLib. We folded some into CoreLib in the past. We definitely need CoreLib too and it's already include above.

Comment (marek-safar) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:190]: I see so there is a circular dependency between these and SPC?

Comment (akoeplinger) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:190]: > Is System.Private.Corelib.dll missing in the list?

SPC is already included at line 74.

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props (hunk 1 above)
Comment (akoeplinger) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:183]: awesome, thanks!

**Comment by akoeplinger** [comment]
> Is the AOT compiler going to stay in Microsoft.DotNet.ILCompiler?
> I am wondering whether it would make sense to do the same change for all hosts and targets.

@jkotas the AOT compiler is going to stay in Microsoft.DotNet.ILCompiler, or at least I don't see a reason why we should move it.

We can make the runtime pack change for all hosts/targets, I just started with the new platforms so the risk is lower.
